modules/diskq/logqueue-disk-reliable.c

```diff
@@ -276,6 +281,9 @@ _pop_head(LogQueue *s, LogPathOptions *path_options)
 
   log_queue_queued_messages_dec(s);
 
+  if (qdisk_corrupt)
+    log_queue_disk_restart_corrupted(&self->super);
```

Comment:
Could you explain this fix in detail, please?

What about threading? As I see, this is running in the same thread as `log_queue_disk_read_message()`.

Reply:
About queue emptying: I think we should try sending these messages out before replacing the queue (to avoid message loss).

Reply:
It would be nice to send them out, but I think it is a complex task.

As you know, in case of reliable disk buffer, the memory queue messages are stored with positions, and we only pop a message from them, if we popped its duplicate from the disk, based on its position.
If we start a new disk-buffer file, but keep the messages in the memory queues, this would rarely be true, and if it were, it would be a problem, as the message in the memory queue and the message in the disk are on the same position, but they are different.
So we would need another memory only message queue for the messages left behind from the previous disk-buffer file.

We can do this, but it was never done before, so it is not a regression, and I would not prioritize it high.

Reply:
Right, okay.